baseframe/forms/helper.py

```diff
@@ -0,0 +1,55 @@
+import six
+from six.moves.urllib.parse import quote as urlquote
+from .. import asset_cache
+from mxsniff import mxsniff, MXLookupException
+
+__all__ = ['is_public_email_domain']
+
+
+def is_public_email_domain(email_or_domain, timeout=30):
+    """
+    Checks if the given email address or domain belongs to a public email domain
+
+    :param email_or_domain: email address or domain name to check
+    :param timeout: seconds after which mxsniff will give up retrieving MX data
```

Comment:
"Lookup timeout (will raise an exception if timeout happens before the check is completed and no default is provided)"